src/todo.js

```diff
@@ -0,0 +1,170 @@
+import Task from './task';
+import allowNewline from './allow-newline';
+
+export default class {
+  constructor() {
+    this.tasks = localStorage.tasks ? JSON.parse(localStorage.tasks) : [];
+  }
+
+  addTask(description = '') {
+    const newEntry = document.getElementById('new-item');
+    if (description || newEntry.value) {
+      if (!description) description = newEntry.value;
+      newEntry.value = '';
+      this.tasks.push(new Task(description, this.tasks.length));
+      this.updateLocalStorage();
+      this.renderList();
+    }
+  }
+
+  updateIndexes() {
+    let i = 0;
+    this.tasks.forEach((task) => {
+      task.index = i;
+      i += 1;
+    });
+    this.updateLocalStorage();
+    this.renderList();
+  }
+
+  deleteTask(index) {
+    this.tasks.splice(index, 1);
+    this.updateIndexes();
+  }
+
+  clearAllBtn() {
+    this.tasks = this.tasks.filter((task) => !task.completed);
+    this.updateIndexes();
+  }
+
+  updateLocalStorage() {
+    localStorage.tasks = JSON.stringify(this.tasks);
+  }
+
+  renderList() {
+    const list = document.getElementById('my-list');
+    list.innerHTML = '';
+    this.tasks.forEach((task) => {
+      list.innerHTML += `
+        <li id="task-${task.index}">
+          <input type="checkbox" class="checkbox" checked="${task.completed}">
+          <div class="description${task.completed ? ' completed' : ''}" contenteditable="true">${task.description}</div>
+          <button class="li-btn bin hidden"></button>
+          <button class="li-btn v-dots"></button>
+        </li>
+      `;
+    });
+    if (this.tasks) {
+      list.innerHTML += `
+        <li>
+          <a class="clear-completed">Clear all completed</a>
+        </li>
+      `;
+    }
+    this.addEventListeners();
+  }
+
+  addEventListeners() {
+    const obj = this;
+
+    function toggleCompleted(e) {
+      const index = e.target.parentElement.id.split('-')[1];
+      obj.tasks[index].completed = !obj.tasks[index].completed;
+      document.getElementById(`task-${index}`).children[1].classList.toggle('completed');
+      obj.updateLocalStorage();
+      // console.table(JSON.parse(localStorage.tasks));
+    }
+
+    function toggleEditMode(parent) {
+      parent.classList.toggle('edit-mode');
+      parent.children[2].classList.toggle('hidden');
+      parent.children[3].classList.toggle('hidden');
+    }
+
+    function dragTask(index) {
+      const task = document.getElementById(`task-${index}`);
+      task.draggable = true;
+      task.addEventListener('dragstart', () => {
+        task.classList.add('dragging');
+      });
+      task.addEventListener('dragend', () => {
+        task.classList.remove('dragging');
+      });
+    }
+
+    /* eslint-disable no-restricted-syntax, eqeqeq */
+    function dropTask(index) {
+      const task = document.getElementById(`task-${index}`);
```

Comment:
Using two same lines of code many times, in many functions, makes your lines of code many, and also it makes your code DRY since they are duplicates which is what you do not want to have 😞 . You can try to minimize these variables on line (85 & 97) into one global variable, which can be reusable 🔂anywhere you wish to use it 😀.

Reply:
### Hello, @Ntwali-Josue!
Thanks for pointing that out this issue! :pray:  I will definitely work on a better solution for that issue! :+1: